gridscale/resource_gridscale_storage.go

```diff
@@ -273,6 +274,21 @@ func resourceGridscaleStorageCreate(d *schema.ResourceData, meta interface{}) er
 
 	//since only one template can be used, we can just look at index 0
 	if _, ok := d.GetOk("template"); ok {
+		_, okPwd := d.GetOk("template.0.password")
+		_, okPwdType := d.GetOk("template.0.password_type")
+		_, okSSH := d.GetOk("template.0.sshkeys")
+		//return error when both password and sshkey are not set
+		if !okPwd && !okSSH {
+			return fmt.Errorf("password or SSH key must be set")
+		}
+		//return error when both password and sshkey are set
+		if okPwd && okSSH {
```

Comment:
@bkircher what do you think about allowing both (both password and sshkey) or either of them?

Reply:
According to API docs its either("Creating a storage with a template is required either sshkey or password") and error message also indicates it's either.